barretenberg/cpp/src/barretenberg/plonk_honk_shared/composer/composer_lib.hpp

```diff
@@ -53,9 +53,6 @@ void construct_lookup_read_counts(typename Flavor::Polynomial& read_counts,
 {
     // TODO(https://github.com/AztecProtocol/barretenberg/issues/1033): construct tables and counts at top of trace
     size_t offset = dyadic_circuit_size - circuit.get_tables_size();
-    read_counts = typename Flavor::Polynomial(circuit.get_tables_size(), dyadic_circuit_size, /*start index*/ offset);
```

Comment:
This is NOT safe to do - Its not guaranteed that all values of read_counts/tags are set below. My suggestion was to remove the TODO, not the initialization

Reply:
or maybe these polys are being initialized elsewhere? in that case that would be fine. In general the code below will only set a small subset of the poly coeffs so it would not be safe to not init the entire block to zero

Reply:
Yeah I wasn't really thinking redundant - but it's weird to be inconsistent about how we down-size these polys, so I was thinking it could all be done in one place perhaps